File: frameworks/runtime-src/Framework/net/MWNetProtocol.cpp
#include "MWNetProtocol.h"

#include "MWNetHandler.h"

#include <new>

using namespace std;

MW_FRAMEWORK_BEGIN

MWNetProtocol *MWNetProtocol::create(const std::string &protocolId, const std::string &serviceId, mwframework::MWNetHandler *handler)
{
    auto pRet = new (nothrow) MWNetProtocol();
    if (pRet && pRet->init(protocolId, serviceId, handler)) {
        pRet->autorelease();
        return pRet;
    }
    CC_SAFE_RELEASE(pRet);
    return nullptr;
}

bool MWNetProtocol::init(const std::string &protocolId, const std::string &serviceId, mwframework::MWNetHandler *handler)
{
    if (protocolId.size() <= 0 || serviceId.size() <= 0 || !handler) {
        return false;
    }
    
    _protocolId = protocolId;
    _serviceId = serviceId;
    _relatedHandler = handler;
    _relatedHandler->retain();
    
    return true;
}

MWNetProtocol::~MWNetProtocol()
{
    CC_SAFE_RELEASE(_relatedHandler);
}

MW_FRAMEWORK_END

File: frameworks/runtime-src/Framework/lua/auto/api/MWNetProtocol.lua

--------------------------------
-- @module MWNetProtocol
-- @extend MWObject,IUninheritable<mwframework::MWNetProtocol>
-- @parent_module mw

--------------------------------
-- 
-- @function [parent=#MWNetProtocol] getRelatedHandler 
-- @param self
-- @return MWNetHandler#MWNetHandler ret (return value: mw.MWNetHandler)
        
--------------------------------
-- 
-- @function [parent=#MWNetProtocol] getServiceId 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#MWNetProtocol] getProtocolId 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#MWNetProtocol] setServiceId 
-- @param self
-- @param #string var
        
--------------------------------
-- 
-- @function [parent=#MWNetProtocol] setProtocolId 
-- @param self
-- @param #string var
        
--------------------------------
-- Create a new NetProtocol.<br>
-- param protocolId Protocol id.<br>
-- param serviceId Service id.<br>
-- param handler Related net handler.<br>
-- return Return a new NetProtocol object.
-- @function [parent=#MWNetProtocol] create 
-- @param self
-- @param #string protocolId
-- @param #string serviceId
-- @param #mw.MWNetHandler handler
-- @return MWNetProtocol#MWNetProtocol ret (return value: mw.MWNetProtocol)
        
return nil
